### DISCUSSION TITLE
Reading MEP protocol

### DISCUSSION THREAD

**Discussion by Jellybeanz1980** [UtilitechAS/amsreader-firmware#697] [General]
Is reading MEP protocol schedules/planned?  EHorvat has already implemented it, but I'd like a more "native" implementation :)

**Comment by gskjold** [comment]
If you have any documentation for it, I can have a look at least

**Reply by Jellybeanz1980** [reply]
The documentation is in https://github.com/OSGP-Alliance-MEP-and-Optical/Documentation . The document is huge and not easy to understand which approach to take, and impossible to implement without a meter to test/debug on.

The easiest way would be look at the implementation  in the github link https://github.com/DabblerDK/MEP-SW-ESP32, https://github.com/DabblerDK/MEP-SW-ESP32/blob/main/NES-MEP-UI/NES-MEP.cpp to be more exact.

**Comment by ArnieO** [comment]
https://github.com/ehorvat1/NES-MEP-Reader

**Reply by gskjold** [reply]
I was looking for a more general documentation :)

**Reply by ArnieO** [reply]
I think there is something more in a link on that repo: https://github.com/OSGP-Alliance-MEP-and-Optical

Last time i read about this (IIRC; this is quite some time ago), the Danish company Dabbler (https://www.dabbler.dk) had made some kind of agreement allowing them to implement the MEP protocol. So you may find some information on their website.

**Reply by Allan-Hoejberg** [reply]
Hi Guys, any further thoughts about supporting the Echelon/NES Smart Meters ?

The option from Dabbler.dk is no longer valid as they no longer have the smartmeter to test with.
Ustepper, use to sell an ESP32-C3 based modul that fits directly into the meter

So I think the main opstricel is decoding the MEP protocol

**Reply by ArnieO** [reply]
We think there are too few users to justify the development cost.
